lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp

```diff
@@ -2875,4 +2875,64 @@ void mlir::torch::onnx_c::populateDefaultDomainQtoZ(
                 /*Torch_BoolType:$antialias*/ cstFalse);
         return success();
       });
+  patterns.onOp(
+      "Upsample", 9, [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+        Torch::ValueTensorType resultType;
+        std::string mode;
+        Value input, scales;
+        if (binder.tensorOperands(input, scales) ||
+            binder.customOpNameStringAttr(mode, "mode", "nearest") ||
+            binder.tensorResultType(resultType)) {
+          return failure();
+        }
+
+        if (mode != "nearest")
+          return rewriter.notifyMatchFailure(
+              binder.op, "unsupported interpolation mode other than nearest");
+
+        int64_t resultRank = resultType.getSizes().size();
+        if (resultRank != 4)
+          return rewriter.notifyMatchFailure(binder.op,
+                                             "supports 2d sampling only");
+
+        SmallVector<Value> outputSizeIntList;
+        if (resultType.areAllSizesKnown()) {
+          SmallVector<int64_t> resultShape(resultType.getSizes());
+          for (uint32_t i = 2; i < resultShape.size(); i++)
+            outputSizeIntList.push_back(rewriter.create<Torch::ConstantIntOp>(
+                binder.getLoc(), rewriter.getI64IntegerAttr(resultShape[i])));
+        } else {
+          Torch::BaseTensorType scalesType =
+              cast<Torch::BaseTensorType>(scales.getType());
+          SmallVector<int64_t> selectSizes{1};
+          Type selectResultType = scalesType.getWithSizesAndDtype(
+              selectSizes, scalesType.getOptionalDtype());
+          Value zero = rewriter.create<Torch::ConstantIntOp>(
```

Comment:
I'd prefer the use of the shorter build form for this constant int op, since specifying rewriter.getType<Torch::IntType>() is superfluous.